src/databricks/labs/ucx/source_code/python_linter.py

```diff
@@ -312,21 +315,33 @@ def list_dbutils_notebook_run_calls(linter: ASTLinter) -> list[NotebookRunCall]:
         return [NotebookRunCall(call) for call in calls]
 
     @staticmethod
-    def list_import_sources(linter: ASTLinter) -> list[ImportSource]:
-        # TODO: make this code more robust, because it fails detecting imports on UCX codebase
+    def list_import_sources(linter: ASTLinter, problem_type: P) -> tuple[list[ImportSource], list[P]]:
```

Comment:
```suggestion
    def list_import_sources(linter: ASTLinter) -> tuple[list[ImportSource], list[DependencyProblem]]:
```
i don't believe generic type is necessary here.

Reply:
It avoids a circular import